### PR TITLE
polygon deployment

### PR DESCRIPTION
**Pull request by jellegerbrandy**

### CODE REVIEW

File: scripts/deployments/hatvaultsregistry-deploy.js [hunk 1]
```diff
@@ -96,9 +103,16 @@ async function main(config) {
...
     }
-
-    return { hatVaultsRegistry, rewardController, rewardControllerImplementation, hatVaultImplementation, arbitrator };
-}
+    return { 
+        hatVaultsRegistry, 
+        rewardControllers: rewardControllers,
+        rewardControllerImplementations: rewardControllerImplementations,
+        rewardController: rewardControllers[0], // return this value to keep it backward compatible
+        rewardControllerImplementation: rewardControllerImplementations[0], //return this val for backwards compatilibity
```
Comment (ben-kaufman) [inline scripts/deployments/hatvaultsregistry-deploy.js:111]: Why keep backward compatibility here though? I think better to just remove

File: scripts/deployments/hatvaultsregistry-deploy.js [hunk 2]
```diff
@@ -37,31 +37,38 @@ async function main(config) {
...
-    const rewardController = await upgrades.deployProxy(RewardController, [
-        hatToken,
-        governance,
-        startBlock,
-        epochLength,
-        epochRewardPerBlock
-    ]);
+    const rewardControllers = [];
+    const rewardControllerImplementations = [];
+    for (const rewardControllerConfig of config.rewardControllers) {
```
Comment (lirona) [inline scripts/deployments/hatvaultsregistry-deploy.js:44]: for (const rewardControllerConfig of config["rewardControllers"]) {
 for consistency :)

**Review by lirona — COMMENTED** [review]
looks good, maybe we should have a test with initialization with more then one reward controller then..

File: scripts/deployments/hats-deploy.js [hunk 3]
```diff
@@ -65,8 +65,8 @@ async function main() {
     hatTokenLock: config["hatTokenLock"],
     tokenLockFactory: config["tokenLockFactory"],
     hatVaultsRegistry: config["hatVaultsRegistry"],
-    rewardController: config["rewardController"],
-    rewardControllerImplementation: config["rewardControllerImplementation"],
+    rewardControllers: config["rewardControllers"],
+    rewardControllerImplementations: config["rewardControllerImplementations"],
```
[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply scripts/deployments/hats-deploy.js:69]: i'm not getting it, what do you mean?

**Pull request merged**